invenio_rdm_migrator/load/postgresql/table_generators/single.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Single table generator module."""
+
+from .base import CSVTableGenerator, MemoryTableGenerator
+
+
+class CSVSingleTableGenerator(CSVTableGenerator):
+    """Contains only one table to yield."""
+
+    def __init__(self, table, pks=None, post_load_hooks=None):
+        """Constructor."""
+        assert not isinstance(table, (list, dict))
+
+        super().__init__(tables=[table], pks=pks, post_load_hooks=post_load_hooks)
+
+    def _generate_rows(self, data, **kwargs):
+        """Yield generated rows."""
+        table = self.tables[0]
+        yield table(**data)
+
+
+# FIXME: feels wrong to duplicate absolutely all functions, use a Mixin?
```

Comment:
Note this comment. In general I think we need to re-design a bit the table generators:
1- Naming: they are becoming long java-like monsters
2- We are over-subclassing just to change tiny things (it's not fully wrong, since that's what specializing is for but there might be simpler/cleaner ways)